src/state/medicaltypes/actions.ts

```diff
@@ -0,0 +1,198 @@
+import isEmpty from "lodash.isempty";
+import { Dispatch } from "redux";
+import {
+  Configuration,
+  MedicalTypeControllerApi,
+  MedicalTypeDTO,
+  CreateMedicalTypeUsingPOSTRequest,
+  UpdateMedicalTypeUsingPUTRequest,
+  OperationOpts,
+} from "../../generated";
+import { applyTokenMiddleware } from "../../libraries/apiUtils/applyTokenMiddleware";
+import { IAction } from "../types";
+import {
+  GET_MEDICALTYPE_FAIL,
+  GET_MEDICALTYPE_LOADING,
+  GET_MEDICALTYPE_SUCCESS,
+  CREATE_MEDICALTYPE_FAIL,
+  CREATE_MEDICALTYPE_LOADING,
+  CREATE_MEDICALTYPE_RESET,
+  CREATE_MEDICALTYPE_SUCCESS,
+  EDIT_MEDICALTYPE_LOADING,
+  EDIT_MEDICALTYPE_FAIL,
+  EDIT_MEDICALTYPE_RESET,
+  EDIT_MEDICALTYPE_SUCCESS,
+  DELETE_MEDICALTYPE_LOADING,
+  DELETE_MEDICALTYPE_FAIL,
+  DELETE_MEDICALTYPE_RESET,
+  DELETE_MEDICALTYPE_SUCCESS,
+} from "./consts";
+
+const medicalTypeControllerApi = new MedicalTypeControllerApi(
+  new Configuration({ middleware: [applyTokenMiddleware] })
+);
+
+export const newMedicalType =
+  ( medicalType: MedicalTypeDTO) => //, ignSimilar: boolean
+  (dispatch: Dispatch<IAction<null, {}>>): void => {
+    dispatch({
+      type: CREATE_MEDICALTYPE_LOADING,
+    });
+
+    var newMedicalTypeRequest: CreateMedicalTypeUsingPOSTRequest = { medicalTypeDTO: medicalType }
+    medicalTypeControllerApi.createMedicalTypeUsingPOST(newMedicalTypeRequest).subscribe( 
```

Comment:
For these two lines you can just simplify like: 
` medicalTypeControllerApi.createMedicalTypeUsingPOST({ medicalTypeDTO: medicalType }).subscribe(...`
and then remove the variable declaration.

Reply:
Any way I think, since this newMedicalType action is not used now we have to remove it.